conan/tools/microsoft/visual.py

```diff
@@ -89,9 +89,14 @@ def msvc_runtime_flag(conanfile):
     runtime = settings.get_safe("compiler.runtime")
     if compiler == "Visual Studio":
         return runtime
-    if compiler == "msvc" or compiler == "intel-cc":
+    if runtime is not None:
+        if runtime == "static":
+            runtime = "MT"
+        elif runtime == "dynamic":
+            runtime = "MD"
+        else:
+            raise ConanException("compiler.runtime should be 'static' or 'dynamic'")
```

Comment:
So why do you still allow MD, MT, MTd, MDd in settings.yml for clang? It will break users still using these runtime values.

Reply:
Something like this:

```
    # old model
    if compiler == "Visual Studio" or (compiler == "clang" and runtime in ["MD", "MDd", "MT", "MTd"]):
        return runtime
    # new model
    if runtime is not None:
        if runtime == "static":
            runtime = "MT"
        elif runtime == "dynamic":
            runtime = "MD"
        else:
            raise ConanException("compiler.runtime should be 'static' or 'dynamic'")
```

Reply:
It is protected earlier with 
```python
if compiler == "Visual Studio":
        return runtime
```

It will only reach this point for other compilers (including ``msvc``), which should be safe to assume they can only use static/dynamic runtime. But this method shouldn't' allow ``MD/MT...`` definitions to be forward compatible with 2.0

Reply:
It's not protected against `clang`. It's the point of my comment.

Reply:
I was thinking forward to 2.0, but ok, lets support legacy settings in this new helper. https://github.com/conan-io/conan/pull/10920 should fix it, please check